src/modules/historical.ts

```diff
@@ -70,10 +70,18 @@ export default function historical(
           const value = queryOptions[fieldName];
           if (value instanceof Date)
             queryOptions[fieldName] = Math.floor(value.getTime() / 1000);
-          else if (typeof value === "string")
-            queryOptions[fieldName] = Math.floor(
-              new Date(value as string).getTime() / 1000
-            );
+          else if (typeof value === "string") {
+            const timestamp = new Date(value as string).getTime();
+
+            if (isNaN(timestamp) || timestamp <= 0)
+              throw new Error(
+                "yahooFinance.historical() option '" +
+                  fieldName +
+                  "' invalid date provided."
+              );
```

Comment:
This is great but we can help the user even more by letting them know what the troublesome input was, i.e. `"' invalid date provided: '" + value + "'"` or something.